packages/cloud_functions/cloud_functions/ios/Classes/FLTFirebaseFunctionsPlugin.m

```diff
@@ -93,7 +94,17 @@ - (void)httpsFunctionCall:(id)arguments withMethodCallResult:(FLTFirebaseMethodC
     [functions useEmulatorWithHost:[url host] port:[[url port] intValue]];
   }
 
-  FIRHTTPSCallable *function = [functions HTTPSCallableWithName:functionName];
+  FIRHTTPSCallable *function;
+
+  if (![functionName isEqual:[NSNull null]]) {
+    function = [functions HTTPSCallableWithName:functionName];
+  } else if (![functionUri isEqual:[NSNull null]]) {
+    function = [functions HTTPSCallableWithURL:[NSURL URLWithString:functionUri]];
+  } else {
+    result.error(@"IllegalArgumentException", @"Either functionName or functionUri must be set",
```

Comment:
Still not 100% sure on this. Shouldn't we capture on android side the illegal argument exception and have an error code in the same format as other exception codes? (e.g. something like: `illegal-argument-exception`) 🤔 . Then we can match it here with the same exception code.

Reply:
Huum, for an internal error like this one which does not depend on input from the user (since the user can only provide either name or uri) I'm not sure if it's worth it (already caught in assert on the dart side)
It was more to prevent having a pending result. This error was more for future developers to see the case properly.

Reply:
That's true actually, LGTM then 👍

Reply:
So my understanding is this should never happen since dart code prevents it?

Reply:
Yes exactly 👍